apps/general/models.py

```diff
@@ -1,3 +1,230 @@
+        super().save(*args, **kwargs)
+
+    def verify_e_way_password(self, password):
+        # Verify e_way_password using bcrypt
+        if self.e_way_password:
+            return bcrypt.checkpw(password.encode(), self.e_way_password)
+        return False
+
+    def verify_gstn_password(self, password):
+        # Verify gstn_password using bcrypt
+        if self.gstn_password:
+            return bcrypt.checkpw(password.encode(), self.gstn_password)
+        return False
+    
+
+# Dummy encryption and decryption functions for demonstration purposes
+def encrypt(text):
+    # Encode the text using base64
+    encoded_bytes = base64.b64encode(text.encode("utf-8"))
+    encrypted_text = encoded_bytes.decode("utf-8")
+    return encrypted_text
+
+def decrypt(encrypted_text):
+    # Decode the text using base64
+    decoded_bytes = base64.b64decode(encrypted_text.encode("utf-8"))
+    decrypted_text = decoded_bytes.decode("utf-8")
+    return decrypted_text
+
+class EncryptedTextField(models.TextField):
+    """
+    A custom field to store encrypted text.
+    """
+    def from_db_value(self, value, expression, connection):
+        if value is None:
+            return value
+        # Implement decryption logic here
+        return decrypt(value)
+
+    def to_python(self, value):
+        if isinstance(value, str):
+            # Implement decryption logic here
+            return decrypt(value)
+        return value
+
+    def get_prep_value(self, value):
+        # Implement encryption logic here
+        return encrypt(value)
+
+class BranchBankDetails(models.Model):
+    bank_detail_id = models.AutoField(primary_key=True)
+    branch_id = models.ForeignKey(Branches, on_delete=models.CASCADE, null=True, default=None, db_column = 'branch_id')
+    bank_name = models.CharField(max_length=255)
+    account_number = EncryptedTextField(max_length=255)  # Using custom encrypted field
+    branch_name = models.CharField(max_length=255, default=None, null=True)
+    ifsc_code = models.CharField(max_length=100, default=None, null=True)
+    swift_code = models.CharField(max_length=100, default=None, null=True)
+    address = models.TextField(default=None, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f"Bank details for Branch ID: {self.bank_name}"
+
+    class Meta:
+        db_table = 'branch_bank_details' 
+
+'''
+#If you want to decrypt then you can uncomment this and run... in output you will find the decrypted account number 
+
+import base64
+
+# Encoded account number
+encoded_account_number = "pass encoded account number here which will be available in your database"
+
+# Decode from base64
+decoded_bytes = base64.b64decode(encoded_account_number)
+
+# Convert bytes to string
+original_account_number = decoded_bytes.decode("utf-8")
+
+print("Decrypted Account Number:", original_account_number)
+
+'''
```

Comment:
Please remove above commented code.

Reply:
Yeah Thulasi, If I remove this when you run server  it will print the original account number everytime. So that's why I have commented.